api/v1beta1/validations.go

```diff
@@ -52,19 +52,19 @@ func GetIntOrPercentValueSafely(intOrStr *intstr.IntOrString) (int, bool, error)
 		return intOrStr.IntValue(), false, nil
 	case intstr.String:
 		s := intOrStr.StrVal
+		isPercent := false
 
 		if strings.HasSuffix(s, "%") {
```

Comment:
`foo%` matches this condition but is not an expected percentage value, as well as `101%`. Do we consider that it is ok (the `foo%` will generate an error when casting the value to an integer anyway but the `101%` will still match) or should we validate that through a regex instead?

Reply:
I consider that ok. `foo%` will get the right return value, imo `0, false, error`. `101%` is a valid percentage, just not for `count`. That's on the caller to care about (and our callers all do, i believe).